src/js/components/Select/SelectContainer.js

```diff
@@ -483,6 +490,9 @@ const SelectContainer = forwardRef(
                       onMouseOver={
                         !optionDisabled ? onActiveOption(index) : undefined
                       }
+                      onMouseOut={
+                        !optionDisabled ? onResetActiveOption(index) : undefined
```

Comment:
Instead of adding `onResetActiveOption` what if we just did something like this instead?
```suggestion
                        !optionDisabled ? onActiveOption(-1) : undefined
```